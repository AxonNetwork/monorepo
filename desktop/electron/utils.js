const path = require('path')

module.exports.getAppPath = function() {
    try {
        return require('electron').app.getAppPath()
    } catch (err) {
        return require('electron').remote.app.getAppPath()
    }
}

module.exports.getEnv = function() {
    const appPath = module.exports.getAppPath()
    const binariesPath = path.join(appPath, '../desktop/build-resources/binaries')
    const env = Object.assign({}, process.env, { CONSCIENCE_APP_PATH: appPath, CONSCIENCE_BINARIES_PATH: binariesPath })
    env.PATH = binariesPath + ':' + env.PATH + ':/usr/local/bin:/usr/bin:/usr/sbin:/sbin'
    return env
}
