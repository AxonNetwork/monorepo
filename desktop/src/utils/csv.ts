export function parseCSV(contents: string){
	const lines = contents.split("\n")
	const data = lines.map(line => line.split(","))
	return data
}