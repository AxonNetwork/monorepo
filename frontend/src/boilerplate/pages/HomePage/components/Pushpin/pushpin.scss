.pushpin {
  height: 825px;
}
