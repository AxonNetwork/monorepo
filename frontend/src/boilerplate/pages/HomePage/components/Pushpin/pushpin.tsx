import * as React from 'react';
import { InjectedI18nProps, InjectedTranslateProps, translate } from 'react-i18next';

import i18ns from './i18n';

const styles = require('./pushpin.scss');

interface IPushpinProps extends InjectedI18nProps, InjectedTranslateProps {
  color: string;
  depth: string;
}

export class Pushpin extends React.Component<IPushpinProps> {
  constructor(props: IPushpinProps) {
    super(props);
    this.loadI18ns();
  }

  public loadI18ns() {
    const { i18n } = this.props;
    Object.keys(i18ns).forEach((key) => {
      i18n.addResourceBundle(key, 'Pushpin', i18ns[key]);
    });
  }

  public render() {
    const { t } = this.props;
    return (
      <div id={this.props.color} className={`${styles.pushpin} ${this.props.color} ${this.props.depth}`}>
        <nav className='pushpin-nav pin-top' data-target={'pushpin-' + this.props.color}>
          <div className={`nav-wrapper ${this.props.color} pushpin-${this.props.color}`}>
            <div className='container'>
              <a href='#' className='brand-logo'>{t(this.props.color)}</a>
              <ul id={`nav-demo-${this.props.color}`} className='right hide-on-med-and-down'>
                <li><a href='#'>{t('link-1')}</a></li>
                <li><a href='#'>{t('link-2')}</a></li>
              </ul>
            </div>
          </div>
        </nav>
      </div>
    );
  }
}

export default translate('Pushpin')(Pushpin);
