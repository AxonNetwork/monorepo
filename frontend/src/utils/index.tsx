export const isProduction = process.env.NODE_ENV === 'production'

const exec = require('child_process').exec

export async function execCmd(cmd: string, cwd: string) {
    return new Promise((resolve, reject) => {
        exec(cmd, { encoding: 'utf8', cwd }, (error: Error|null, stdout: string, stderr: string) => {
            if (error) {
                reject(error)
            } else {
                resolve({ stdout, stderr })
            }
        })
    })
}
