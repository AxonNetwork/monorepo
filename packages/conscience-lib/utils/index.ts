export { default as autobind } from './autobind'
export { default as strToColor } from './strToColor'
export { default as schemes } from './codeColorSchemes'
export { default as getLanguage } from './getLanguage'
export { default as checkVisible } from './checkVisible'
export { default as nonCacheImg } from './nonCacheImg'
export { repoPageToString, stringToRepoPage } from './parseRepoPage'
export { orgPageToString, stringToOrgPage } from './parseOrgPage'
export { getConscienceURI } from './uri'
export { removeEmail, extractEmail } from './email'
export { parseCSV } from './csv'
export { isTextFile } from './fileTypes'
export { getLastVerifiedEvent, getLastVerifiedEventCommit, getLastVerifiedEventFile, getFirstVerifiedEvent } from './timeline'
