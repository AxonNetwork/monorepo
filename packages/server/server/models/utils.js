import { dynamo } from '../config/aws'

export function makeID() {
    let text = ''
    const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'

    for (let i = 0; i < 10; i++) {
        text += possible.charAt(Math.floor(Math.random() * possible.length))
    }

    return text
}

export async function getAll(queryParams) {
    let items = []
    let ExclusiveStartKey

    do {
        const resp = await dynamo.queryAsync({
            ...queryParams,
            ExclusiveStartKey,
        })

        if (resp.Items) {
            items = items.concat(resp.Items)
        }

        ExclusiveStartKey = resp.LastEvaluatedKey
    } while (ExclusiveStartKey)

    return items
}
