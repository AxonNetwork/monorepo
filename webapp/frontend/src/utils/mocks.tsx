// Mock dispatch
export const dispatchMock = () => { return; };

// Mock translation
export const tMock = (k: string | string[]) => k;

// Mock i18n
export const i18nMock: any = {
  addResourceBundle: () => { return; },
};
