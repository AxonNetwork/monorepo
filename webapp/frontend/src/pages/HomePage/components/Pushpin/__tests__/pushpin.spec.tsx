import React from 'react';
import TestRenderer from 'react-test-renderer';

import { i18nMock, tMock } from 'utils/mocks';
import { Pushpin } from '../pushpin';

describe('Pushpin', () => {
  it('should renders correctly', () => {
    const renderer = TestRenderer.create(
      <Pushpin color='red' depth='base' t={tMock} tReady={true} i18n={i18nMock} />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
