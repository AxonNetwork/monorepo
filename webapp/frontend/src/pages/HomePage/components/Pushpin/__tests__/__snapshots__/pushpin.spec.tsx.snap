// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pushpin should renders correctly 1`] = `
<div
  className="pushpin red base"
  id="red"
>
  <nav
    className="pushpin-nav pin-top"
    data-target="pushpin-red"
  >
    <div
      className="nav-wrapper red pushpin-red"
    >
      <div
        className="container"
      >
        <a
          className="brand-logo"
          href="#"
        >
          red
        </a>
        <ul
          className="right hide-on-med-and-down"
          id="nav-demo-red"
        >
          <li>
            <a
              href="#"
            >
              link-1
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              link-2
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
